src/components/Modals/NewCourse/NewCourseModal.vue

```diff
@@ -133,19 +126,27 @@ export default Vue.extend({
             for (let k = 0; k < courses.length; k += 1) {
               for (const [, ids] of Object.entries(courses[k])) {
                 if (ids.includes(selectedCourse.id)) {
-                  relatedRequirements.push({ id: subreqs[j].id, name: subRequirement.name });
+                  if (subRequirement.allowCourseDoubleCounting) {
+                    requirementsThatAllowDoubleCounting.push(subRequirement.name);
+                  } else {
+                    relatedRequirements.push({ id: subreqs[j].id, name: subRequirement.name });
+                  }
                   break;
                 }
               }
             }
           }
-          // potential requirements
+          // potential self-check requirements
           if (subreqs[j].fulfilledBy === 'self-check') {
-            potentialRequirements.push({ id: subreqs[j].id, name: subRequirement.name });
+            if (!subRequirement.allowCourseDoubleCounting) {
+              potentialRequirements.push({ id: subreqs[j].id, name: subRequirement.name });
+            }
           }
+          this.requirementsThatAllowDoubleCounting = requirementsThatAllowDoubleCounting;
           this.relatedRequirements = relatedRequirements;
           this.potentialRequirements = potentialRequirements;
-          this.selectedRequirementIDs = relatedRequirements.map(it => it.id);
+          this.selectedRequirementID =
+            relatedRequirements.length > 0 ? relatedRequirements[0].id : '';
```

Comment:
It's important to note that the algorithm used here is still flaky. It doesn't account for toggleable requirements at all. Fixing it would be the job of another diff.